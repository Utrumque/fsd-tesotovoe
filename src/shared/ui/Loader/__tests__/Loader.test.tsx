import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { Loader } from '../index';

describe('Loader', () => {
  it('должен корректно рендериться', () => {
    const { container } = render(<Loader />);
    expect(container).not.toBeNull();
  });

  it('должен содержать элемент loader', () => {
    const { container } = render(<Loader />);
    const loader = container.querySelector('[data-testid="loader"]');
    expect(loader).not.toBeNull();
  });

  it('должен содержать элемент spinner', () => {
    const { container } = render(<Loader />);
    const spinner = container.querySelector('[data-testid="spinner"]');
    expect(spinner).not.toBeNull();
  });
});
