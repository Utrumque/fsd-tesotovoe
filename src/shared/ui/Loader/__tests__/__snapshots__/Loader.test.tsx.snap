// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loader должен корректно рендериться 1`] = `
<div>
  <div
    data-testid="loader"
  >
    <div
      data-testid="spinner"
    />
  </div>
</div>
`;
